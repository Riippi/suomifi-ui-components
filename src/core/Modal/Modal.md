### Basic Modal

Visibility of the Modal is controlled outside the component. Opening and closing the Modal needs to be explicitly set on desired functionality (primary and secondary buttons, esc key).

Aria props override the primary and secondary button labels for screenreaders and allow more detailed description.

```js
import { useState } from 'react';
import {
  Modal,
  Button,
  Paragraph,
  Text
} from 'suomifi-ui-components';

const [visible, setVisible] = useState(false);

<>
  <Button
    onClick={() => {
      setVisible(!visible);
    }}
  >
    Open modal dialogue
  </Button>
  <Modal
    visible={visible}
    title="Test modal"
    primaryButtonLabel="OK"
    ariaPrimaryButtonLabel="Accept changes and close modal dialogue"
    primaryButtonProps={{ onClick: () => setVisible(!visible) }}
    secondaryButtonLabel="Cancel"
    ariaSecondaryButtonLabel="Discard changes and close modal dialogue"
    secondaryButtonProps={{ onClick: () => setVisible(!visible) }}
    onEscKeyDown={() => setVisible(!visible)}
  >
    <Paragraph>
      <Text>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        Phasellus scelerisque elit a consectetur tempor. Morbi sit
        amet lobortis ipsum. Nunc ac ante ligula. Mauris sem urna,
        feugiat eu faucibus in, vehicula sit amet nibh. Duis non
        egestas enim. Sed pharetra, eros a feugiat porttitor, nisi
        eros dapibus mi, in semper augue erat sit amet nulla. Quisque
        non sapien sem.
      </Text>
    </Paragraph>
  </Modal>
</>;
```

### Modal Small Screen Variant, no portal and no scrolling

Small screen variant takes all available screen space.

Not scrollable modals do not allow content scrolling by default. Disabling scrolling also removes the horizontal divider line above the buttons.

Disabling portal renders content inside the modal component, not as the last DOM node when using a portal. When portals are disabled, z-index of the portal is set to 100.

```js
import { useState } from 'react';
import {
  Modal,
  Button,
  Paragraph,
  Text
} from 'suomifi-ui-components';

const [visible, setVisible] = useState(false);

<>
  <Button
    onClick={() => {
      setVisible(!visible);
    }}
  >
    Open modal dialogue
  </Button>
  <Modal
    visible={visible}
    scrollable={false}
    usePortal={false}
    variant="smallScreen"
    title="Test modal"
    primaryButtonLabel="OK"
    primaryButtonProps={{ onClick: () => setVisible(!visible) }}
    onEscKeyDown={() => setVisible(!visible)}
  >
    <Paragraph>
      <Text>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        Phasellus scelerisque elit a consectetur tempor. Morbi sit
        amet lobortis ipsum. Nunc ac ante ligula. Mauris sem urna,
        feugiat eu faucibus in, vehicula sit amet nibh. Duis non
        egestas enim. Sed pharetra, eros a feugiat porttitor, nisi
        eros dapibus mi, in semper augue erat sit amet nulla. Quisque
        non sapien sem.
      </Text>
    </Paragraph>
  </Modal>
</>;
```

### Modal with onClose and onOpen focus set

By default, Modal will focus the first focusable element inside the modal when opened and return the focus to where it was before opening the modal.

_focusOnOpenRef_ and _focusOnCloseRef_ props can be used to change this behaviour.

```js
import { useState, useRef } from 'react';
import {
  Modal,
  Button,
  Paragraph,
  Text
} from 'suomifi-ui-components';

const [visible, setVisible] = useState(false);

const returnFocusRef = useRef(null);
const initialFocusRef = useRef(null);

<>
  <Button
    onClick={() => {
      setVisible(!visible);
    }}
  >
    Open modal dialogue
  </Button>
  <input ref={returnFocusRef} />
  <Modal
    title="Test modal"
    primaryButtonLabel="OK"
    visible={visible}
    primaryButtonProps={{ onClick: () => setVisible(!visible) }}
    focusOnOpenRef={initialFocusRef}
    focusOnCloseRef={returnFocusRef}
    onEscKeyDown={() => setVisible(!visible)}
  >
    <Button>Test Button</Button>
    <input ref={initialFocusRef} />
    <Paragraph>
      <Text>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
        Phasellus scelerisque elit a consectetur tempor. Morbi sit
        amet lobortis ipsum. Nunc ac ante ligula. Mauris sem urna,
        feugiat eu faucibus in, vehicula sit amet nibh. Duis non
        egestas enim. Sed pharetra, eros a feugiat porttitor, nisi
        eros dapibus mi, in semper augue erat sit amet nulla. Quisque
        non sapien sem.
      </Text>
    </Paragraph>
  </Modal>
</>;
```
