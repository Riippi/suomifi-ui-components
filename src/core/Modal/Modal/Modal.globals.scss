/**
 * Styles for disabling main content scrolling when masked by Modal.
 * Styled Components does not provide any feasible solutions for global styles for libraries.
 * This file is compiled into main.css and the class is applied to body when modal is visible.
 */
.fi-modal--disable-body-content {
  overflow: hidden;
}
