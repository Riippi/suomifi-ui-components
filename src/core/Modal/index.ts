export { Modal, ModalProps } from './Modal/Modal';
export { ModalContent, ModalContentProps } from './ModalContent/ModalContent';
export { ModalTitle, ModalTitleProps } from './ModalTitle/ModalTitle';
export { ModalFooter, ModalFooterProps } from './ModalFooter/ModalFooter';
export { ModalButton, ModalButtonProps } from './ModalButton/ModalButton';
