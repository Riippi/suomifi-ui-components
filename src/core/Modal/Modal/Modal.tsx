import React, { Component, ReactNode, createRef } from 'react';
import { default as styled } from 'styled-components';
import { default as ReactModal } from 'react-modal';
import classnames from 'classnames';
import { ModalContent, ModalFooter } from '../';
/* Styles for disabling scrolling. See file for more info. */
import './Modal.globals.scss';
import { baseStyles } from './Modal.baseStyles';

export type ModalVariant = 'smallScreen' | 'default';

export interface ModalProps {
  /** Show or hide the modal */
  visible: boolean;
  /** Application root element id for setting aria-hidden when necessary */
  appElementId?: string;
  /** Modal container div classname for custom styling. */
  className?: string;
  /** Modal content wrapper styles */
  style?: React.CSSProperties;
  /** Children */
  children: ModalContent | ModalFooter | ReactNode;
  /**
   * Variant. Use smallScreen for mobile and small displays
   * @default 'default'
   */
  variant?: ModalVariant;
  /**
   * Show vertical scroll bar if needed and show horizontal divider between content and footer.
   * @default true
   */
  scrollable?: boolean;
  /** Focusable element ref when modal is opened. If not provided, modal title is focused. */
  focusOnOpenRef?: React.RefObject<any>;
  /** Focusable element ref when modal is closed. If not provided, previously focused element will regain focus. */
  focusOnCloseRef?: React.RefObject<any>;
  /** Callback for handling esc key press, e.g. close modal */
  onEscKeyDown?: () => void;
}

interface InternalModalProps extends ModalProps {
  /**
   * Used to capture user provided classname for use with BaseModal.
   * Passed on to the BaseModal inner element with fi-modal classname to allow custom styles.
   * className is internally reserved for BaseModal root element to allow styled component style injection.
   */
  propClassName?: string;
}

export interface ModalProviderState {
  titleRef: React.RefObject<HTMLHeadElement> | null;
  focusTitleOnOpen: boolean;
  variant: ModalVariant;
  scrollable: boolean;
}

const defaultProviderValue: ModalProviderState = {
  /** Focus title on open for resolving title tab index */
  focusTitleOnOpen: true,
  /** Modal title ref for focusing */
  titleRef: null,
  /** Modal's smallScreen setting */
  variant: 'default',
  /** If modal should have scrollable content and size */
  scrollable: true,
};

const {
  Provider: ModalProvider,
  Consumer: ModalConsumer,
} = React.createContext(defaultProviderValue);

export const baseClassName = 'fi-modal';
const modalClassNames = {
  disableBodyContent: `${baseClassName}--disable-body-content`,
  smallScreen: `${baseClassName}--small-screen`,
  noScroll: `${baseClassName}--no-scroll`,
  overlay: `${baseClassName}_overlay`,
  base: `${baseClassName}_base`,
  contentContainer: `${baseClassName}_content-container`,
};

class BaseModal extends Component<InternalModalProps> {
  private titleRef = createRef<HTMLHeadElement>();

  constructor(props: InternalModalProps) {
    super(props);
    if (props.appElementId) {
      ReactModal.setAppElement(`#${props.appElementId}`);
    }
  }

  render() {
    const {
      visible,
      style,
      appElementId,
      propClassName,
      className,
      children,
      variant = 'default',
      scrollable = true,
      focusOnOpenRef,
      focusOnCloseRef = null,
      onEscKeyDown,
    } = this.props;

    return (
      <ReactModal
        bodyOpenClassName={modalClassNames.disableBodyContent}
        portalClassName={classnames(className, modalClassNames.base)}
        overlayClassName={classnames(modalClassNames.overlay)}
        className={classnames(propClassName, baseClassName, {
          [modalClassNames.smallScreen]: variant === 'smallScreen',
          [modalClassNames.noScroll]: scrollable === false,
        })}
        ariaHideApp={!!appElementId}
        isOpen={visible}
        onAfterOpen={() => {
          if (!!focusOnOpenRef && !!focusOnOpenRef.current) {
            focusOnOpenRef.current.focus();
          } else if (!!this.titleRef && !!this.titleRef.current) {
            this.titleRef.current.focus();
          }
        }}
        onAfterClose={() => {
          if (!!focusOnCloseRef && !!focusOnCloseRef.current) {
            focusOnCloseRef.current.focus();
          }
        }}
        onRequestClose={() => {
          if (!!onEscKeyDown) onEscKeyDown();
        }}
        shouldFocusAfterRender={true}
        shouldReturnFocusAfterClose={!focusOnCloseRef}
        shouldCloseOnOverlayClick={false}
        shouldCloseOnEsc={true}
        {...(!!style ? { style: { content: { ...style } } } : {})}
      >
        <ModalProvider
          value={{
            focusTitleOnOpen: !focusOnOpenRef,
            titleRef: this.titleRef,
            variant,
            scrollable,
          }}
        >
          {children}
        </ModalProvider>
      </ReactModal>
    );
  }
}

const StyledModal = styled(BaseModal)`
  ${baseStyles}
`;

/**
 * <i class="semantics" />
 * Use for showing modal content.
 * NOTE: Modal hides the application root node from screen readers using the provided appElementId.
 */
export class Modal extends Component<ModalProps> {
  render() {
    const { className: propClassName, ...passProps } = this.props;
    return <StyledModal propClassName={propClassName} {...passProps} />;
  }
}

export { ModalProvider, ModalConsumer };
