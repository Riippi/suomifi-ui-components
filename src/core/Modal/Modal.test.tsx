import React from 'react';
import { render } from '@testing-library/react';
import { axeTest } from '../../utils/test/axe';

import { Modal } from './Modal';

describe('Basic modal', () => {
  const BasicModal = (
    <Modal
      title="Test modal"
      primaryButtonLabel="OK"
      visible={true}
      style={{ height: '500px' }}
    >
      <p>
        Im baby cold-pressed selfies edison bulb, hot chicken master cleanse
        hoodie tote bag 3 wolf moon yuccie shoreditch quinoa put a bird on it.
        XOXO lo-fi man bun raclette vice thundercats hoodie ethical godard
        master cleanse 8-bit chillwave church-key pok pok. Food truck beard
        bushwick four loko, flexitarian ugh mixtape tattooed jianbing tbh irony
        pitchfork small batch bitters. Hoodie DIY man braid kale chips,
        gochujang trust fund neutra edison bulb roof party single-origin coffee
        iceland. Direct trade heirloom cliche tote bag YOLO. Wolf mumblecore
        lumbersexual tattooed ethical authentic. Kale chips post-ironic
        sartorial ugh mustache helvetica fashion axe, vinyl wolf neutra jean
        shorts.
      </p>
    </Modal>
  );
  it('should match snapshot', () => {
    const { container } = render(BasicModal);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('should not have basic accessibility issues', axeTest(BasicModal));
});
