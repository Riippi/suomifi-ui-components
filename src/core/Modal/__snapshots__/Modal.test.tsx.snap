// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic modal should match snapshot 1`] = `
.c4 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  overflow: visible;
  text-transform: none;
  -webkit-appearance: button;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline-block;
  max-width: 100%;
  cursor: pointer;
}

.c4:-moz-focusring {
  outline: 1px dotted ButtonText;
}

.c4::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.c4::-webkit-file-upload-button {
  -webkit-appearance: button;
  font: inherit;
}

.c4::-webkit-inner-spin-button {
  height: auto;
}

.c4::-webkit-outer-spin-button {
  height: auto;
}

.c4::before,
.c4::after {
  box-sizing: border-box;
}

.c1 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c1::before,
.c1::after {
  box-sizing: border-box;
}

.c2 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c2::before,
.c2::after {
  box-sizing: border-box;
}

.c6 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c6::before,
.c6::after {
  box-sizing: border-box;
}

.c5 {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 600;
  font-size: 14px;
  line-height: 20px;
  padding: 10px 20px;
  min-height: 40px;
  color: hsl(0,0%,100%);
  background: linear-gradient(0deg,hsl(212,63%,37%) 0%,hsl(212,63%,45%) 100%);
  border-radius: 2px;
  text-align: center;
  text-shadow: 0 1px 1px rgba(0,53,122,0.5);
  cursor: pointer;
}

.c5:focus {
  outline: none;
  position: relative;
}

.c5:focus::after {
  content: '';
  position: absolute;
  pointer-events: none;
  top: -2px;
  right: -2px;
  bottom: -2px;
  left: -2px;
  border-radius: 2px;
  background-color: transparent;
  border: 0px solid hsl(0,0%,100%);
  box-sizing: border-box;
  box-shadow: 0 0 0 2px hsl(196,77%,44%);
  z-index: 9999;
}

.c5:hover {
  background: linear-gradient(0deg,hsl(212,63%,45%) 0%,hsl(212,63%,49%) 100%);
}

.c5:active {
  background: hsl(212,63%,37%);
}

.c5.fi-button--disabled,
.c5[disabled],
.c5:disabled {
  text-shadow: 0 1px 1px rgba(41,41,41,0.5);
  background: linear-gradient(0deg,hsl(202,7%,67%) 0%,hsl(202,7%,80%) 100%);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: not-allowed;
}

.c5.fi-button--disabled::after {
  border: none;
  box-shadow: none;
}

.c5.fi-button--fullwidth {
  display: block;
  width: 100%;
}

.c5.fi-button--inverted {
  background: none;
  background-color: hsl(212,63%,45%);
  border: 1px solid hsl(0,0%,100%);
  text-shadow: none;
}

.c5.fi-button--inverted:hover {
  background: linear-gradient(-180deg,rgba(255,255,255,0.1) 0%,rgba(255,255,255,0) 100%);
}

.c5.fi-button--inverted:active {
  background: none;
  background-color: hsl(212,63%,45%);
}

.c5.fi-button--inverted.fi-button--disabled,
.c5.fi-button--inverted[disabled],
.c5.fi-button--inverted:disabled {
  opacity: 0.41;
  background: none;
  background-color: none;
}

.c5.fi-button--secondary {
  color: hsl(212,63%,45%);
  background: none;
  background-color: hsl(0,0%,100%);
  border: 1px solid hsl(212,63%,45%);
  text-shadow: none;
}

.c5.fi-button--secondary:hover {
  background: linear-gradient(0deg,hsl(202,7%,93%) 0%,hsl(0,0%,100%) 100%);
}

.c5.fi-button--secondary:active {
  background: none;
  background-color: hsl(202,7%,93%);
}

.c5.fi-button--secondary.fi-button--disabled,
.c5.fi-button--secondary[disabled],
.c5.fi-button--secondary:disabled {
  color: hsl(202,7%,67%);
  text-shadow: none;
  background: none;
  background-color: hsl(212,63%,98%);
  border-color: hsl(202,7%,67%);
}

.c5.fi-button--secondary-noborder {
  color: hsl(212,63%,45%);
  background: none;
  background-color: hsl(0,0%,100%);
  border: 1px solid hsl(212,63%,45%);
  text-shadow: none;
  border: none;
}

.c5.fi-button--secondary-noborder:hover {
  background: linear-gradient(0deg,hsl(202,7%,93%) 0%,hsl(0,0%,100%) 100%);
}

.c5.fi-button--secondary-noborder:active {
  background: none;
  background-color: hsl(202,7%,93%);
}

.c5.fi-button--secondary-noborder.fi-button--disabled,
.c5.fi-button--secondary-noborder[disabled],
.c5.fi-button--secondary-noborder:disabled {
  color: hsl(202,7%,67%);
  text-shadow: none;
  background: none;
  background-color: hsl(212,63%,98%);
  border-color: hsl(202,7%,67%);
}

.c5.fi-button--tertiary {
  color: hsl(212,63%,45%);
  background: none;
  background-color: hsl(0,0%,100%);
  border: 1px solid hsl(212,63%,45%);
  text-shadow: none;
  background: linear-gradient(0deg,hsl(212,63%,90%),hsl(212,63%,95%));
  border: none;
}

.c5.fi-button--tertiary:hover {
  background: linear-gradient(0deg,hsl(202,7%,93%) 0%,hsl(0,0%,100%) 100%);
}

.c5.fi-button--tertiary:active {
  background: none;
  background-color: hsl(202,7%,93%);
}

.c5.fi-button--tertiary.fi-button--disabled,
.c5.fi-button--tertiary[disabled],
.c5.fi-button--tertiary:disabled {
  color: hsl(202,7%,67%);
  text-shadow: none;
  background: none;
  background-color: hsl(212,63%,98%);
  border-color: hsl(202,7%,67%);
}

.c5.fi-button--tertiary:hover {
  background: hsl(212,63%,95%);
}

.c5.fi-button--tertiary:active {
  background: linear-gradient(0deg,hsl(202,7%,93%),hsl(202,7%,97%));
}

.c5 > .fi-button_icon {
  width: 16px;
  height: 16px;
  margin-right: 8px;
  vertical-align: middle;
  -webkit-transform: translateY(-0.1em);
  -ms-transform: translateY(-0.1em);
  transform: translateY(-0.1em);
}

.c5 > .fi-button_icon.fi-button_icon--right {
  margin-right: 0;
  margin-left: 8px;
}

.c5.fi-button--disabled > .fi-button_icon {
  cursor: not-allowed;
}

.c3 {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
  color: hsl(0,0%,16%);
}

.c3.fi-heading--h1hero {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 40px;
  line-height: 48px;
  font-weight: 600;
}

.c3.fi-heading--h1 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 40px;
  line-height: 48px;
  font-weight: 300;
}

.c3.fi-heading--h2 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 28px;
  line-height: 34px;
  font-weight: 600;
}

.c3.fi-heading--h3 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 22px;
  line-height: 28px;
  font-weight: 600;
}

.c3.fi-heading--h4 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 24px;
  font-weight: 600;
}

.c3.fi-heading--h5 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 600;
}

.c3.fi-heading--h6 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 600;
}

.c3.fi-heading--small-screen {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 32px;
  line-height: 38px;
  font-weight: 300;
}

.c3.fi-heading--small-screen.fi-heading--h1hero {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 32px;
  line-height: 38px;
  font-weight: 600;
}

.c3.fi-heading--small-screen.fi-heading--h1 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 32px;
  line-height: 38px;
  font-weight: 300;
}

.c3.fi-heading--small-screen.fi-heading--h2 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 26px;
  line-height: 32px;
  font-weight: 600;
}

.c3.fi-heading--small-screen.fi-heading--h3 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 20px;
  line-height: 26px;
  font-weight: 600;
}

.c3.fi-heading--small-screen.fi-heading--h4 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 600;
}

.c3.fi-heading--small-screen.fi-heading--h5 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 600;
}

.c3.fi-heading--small-screen.fi-heading--h6 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 20px;
  font-weight: 600;
}

.c0.fi-modal {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 600;
  width: 100vw;
  height: 100vh;
  position: fixed;
  top: 0;
  left: 0;
}

.c0.fi-modal--no-portal {
  z-index: 100;
}

.c0.fi-modal .fi-modal_overlay {
  background-color: rgba(41,41,41,0.5);
  width: 100vw;
  height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0.fi-modal .fi-modal_content-container {
  border-radius: 4px;
  background-color: hsl(0,0%,100%);
  border-top: 4px solid hsl(212,63%,45%);
  height: 50%;
  width: 500px;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c0.fi-modal .fi-modal_content-container .fi-modal_content {
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow-y: auto;
  border-bottom: 1px solid hsl(202,7%,80%);
  padding: 0 30px 30px 30px;
}

.c0.fi-modal .fi-modal_content-container .fi-modal_content .fi-modal_heading {
  padding-top: 24px;
  padding-bottom: 20px;
}

.c0.fi-modal .fi-modal_content-container .fi-modal_content--no-scroll {
  overflow: hidden;
  border-bottom: none;
}

.c0.fi-modal .fi-modal_content-container .fi-modal_footer {
  -webkit-flex: 0 0 80px;
  -ms-flex: 0 0 80px;
  flex: 0 0 80px;
  padding-top: 20px;
  padding-right: 30px;
  padding-bottom: 20px;
  padding-left: 15px;
}

.c0.fi-modal .fi-modal_content-container .fi-modal_footer .fi-modal_button {
  margin-left: 15px;
}

.c0.fi-modal--small-screen .fi-modal_content-container {
  width: 100%;
  height: 100%;
}

.c0.fi-modal--small-screen .fi-modal_content-container .fi-modal_content {
  padding: 0 20px 20px 20px;
}

.c0.fi-modal--small-screen .fi-modal_content-container .fi-modal_content .fi-modal_heading {
  padding: 20px 0;
}

.c0.fi-modal--small-screen .fi-modal_content-container .fi-modal_footer {
  padding-top: 0;
  padding-right: 20px;
  padding-bottom: 15px;
  padding-left: 5px;
}

.c0.fi-modal--small-screen .fi-modal_content-container .fi-modal_footer .fi-modal_button {
  margin-top: 15px;
  margin-left: 15px;
}

<div
  aria-describedby="5_title"
  aria-modal="true"
  class="c0 fi-modal fi-modal--no-portal"
  role="dialog"
>
  <div
    class="c1 fi-modal_overlay"
  >
    <div
      class="c1 fi-modal_content-container"
    >
      <div
        class="c1 fi-modal_content"
      >
        <h2
          class="c2 fi-heading c3 fi-modal_heading fi-heading--h3"
          id="5_title"
        >
          Test modal
        </h2>
        <p>
          Modal Content
        </p>
      </div>
      <div
        class="c1 fi-modal_footer"
      >
        <button
          aria-disabled="false"
          class="c4 fi-button c5 fi-modal_button"
          tabindex="0"
          type="button"
        >
          <span
            aria-hidden="false"
            class="c6"
          >
            OK
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
